std/zip.d

```diff
@@ -1,61 +1,149 @@
-    }
+    // print some data about each member
+    writefln("%10s  %08x  %s", am.expandedSize, am.crc32, name);
+    assert(am.expandedData.length == 0);
+
+    // decompress the archive member
+    zip.expand(am);
+    assert(am.expandedData.length == am.expandedSize);
 }
+---
 
-// Create and write new zip file.
+Example for writing files into a zip archive:
+---
 import std.file : write;
 import std.string : representation;
+import std.zip;
+
+// Create an ArchiveMembers for each file.
+ArchiveMember file1 = new ArchiveMember();
+file1.name = "test1.txt";
+file1.expandedData("Test data.\n".dup.representation);
+file2.compressionMethod = CompressionMethod.none; // don't compress
+
+ArchiveMember file2 = new ArchiveMember();
+file2.name = "test2.txt";
+file2.expandedData("More test data.\n".dup.representation);
+file2.compressionMethod = CompressionMethod.deflate; // compress
+
+// Create an archive and add the member.
+ZipArchive zip = new ZipArchive();
+
+// add ArchiveMembers
+zip.addMember(file1);
+zip.addMember(file2);
+
+// Build the archive
+void[] compressed_data = zip.build();
+
+// Write to a file
+write("test.zip", compressed_data);
+---
+
+$(TABLE
+$(TR $(TD $(LREF ZipArchive))
+     $(TD The whole archive. Some members and properties:
+          $(TABLE
+          $(TR $(TD $(REF1_ALTTEXT $(D this()), ZipArchive.this, std_zip))
+               $(TD Create empty archive.))
+          $(TR $(TD $(REF1_ALTTEXT $(D this(void[])), ZipArchive.this.2, std_zip))
+               $(TD Create archive from a buffer containing the entire contents of an archive.))
+          $(TR $(TD $(REF1_ALTTEXT $(D addMember(ArchiveMember)), ZipArchive.addMember, std_zip))
+               $(TD Add an ArchiveMember to the archive.))
+          $(TR $(TD $(REF1_ALTTEXT $(D deleteMember(ArchiveMember)), ZipArchive.addMember, std_zip))
+               $(TD Delete an ArchiveMember from the archive.))
+          $(TR $(TD $(REF1_ALTTEXT $(D expand(ArchiveMember)), ZipArchive.expand, std_zip))
+               $(TD Expand compressed data of an Archive Member.))
+          $(TR $(TD $(REF1_ALTTEXT $(D build()), ZipArchive.build, std_zip))
+               $(TD Create an array, representing the entire contents of this $(D ZipArchive).))
+          $(TR $(TD $(REF1_ALTTEXT $(D directory), ZipArchive.directory, std_zip))
+               $(TD Associative Array containing all ArchiveMembers.))
+          $(TR $(TD $(REF1_ALTTEXT $(D data), ZipArchive.data, std_zip))
+               $(TD Array, representing the entire contents of this $(D ZipArchive).))
+          $(TR $(TD $(REF1_ALTTEXT $(D comment), ZipArchive.comment, std_zip))
+               $(TD The global comment of the archive file.))
+          $(TR $(TD $(REF1_ALTTEXT $(D isZip64), ZipArchive.comment, std_zip))
+               $(TD True, if $(D ZipArchive) is in zip64 format. Set this to true, if you want to force zip64 format
+                    when using $(D build).))
```

Comment:
Doesn't look good in DDox, though I don't know if the problem is due to the DDoc here or DDox:

http://dtest.dlang.io/artifact/website-2ad4ace80675bb9b91837246ebe69a2b668ec297-59a3acd7ce221d4dcfb2be853c277543/web/library-prerelease/std/zip.html

Reply:
I wrote the table, because Jonathan M Davis suggested that in the forum. But if they are generated automatically in DDox, that's better. I didn't know about DDox though. And indeed the tables don't look good there; it's a little bit of a hack anyway and I wasn't quite happy with the result.

Do you know, when the default view of the documentation will be switched to DDox? From first sight, I'd say the sooner the better.

Reply:
> Do you know, when the default view of the documentation will be switched to DDox? From first sight, I'd say the sooner the better.

Probably no sooner than it getting an active maintainer, which hasn't happened for a few years now.

Reply:
Well, at this point it's more realistic that we'll remove Ddox as the transition never happened :/

Reply:
That's a pity. Ddox looks so much better that ddoc... And it's allready there. Looks like the only thing that's needed is to toggle a switch... What sort of maintainance is needed in addition?